src/utils/schemas.js

```diff
@@ -75,7 +88,38 @@ export function buildDependencies(nodes) {
                                 ...getEnumNames(node.children),
                             },
                         },
-                        ...buildDependencies(node.children),
+                        ...buildDependenciesEnum(node.children),
+                    };
+                }),
+            },
+        },
+    };
+}
+
+/**
+ * @summary Recursively generate `dependencies` for RJSF schema based on tree.
+ * @param {Object[]} nodes - Array of nodes (e.g. `[tree]` or `node.children`)
+ * @returns {{}|{dependencies: {}}}
+ */
+export function buildDependenciesOneOf(nodes) {
```

Comment:
Should we move this to a file named `tree.js`?

Reply:
It makes use of a tree structure, yes, but the result of this function is really only useful in the context of schemas. In my opinion, it is already in a good place.